Translate Render Props (#94)
reactjs/pt-BR.reactjs.org: content/docs/render-props.md

}
```

> Essa abordagem funciona para o nosso caso em específico, mas ainda não atingimos o objetivo de encapsular o comportamento de uma maneira completamente reutilizável. Agora, toda vez que precisarmos da posição do mouse para outro caso, teremos que criar um novo componente (i.e. ou seja, outro `<MouseWithCat>` ) que renderiza algo especificamente para esse caso.

Reviewer: ```suggestion
Essa abordagem funciona para o nosso caso em específico, mas ainda não atingimos o objetivo de encapsular o comportamento de uma maneira completamente reutilizável. Agora, toda vez que precisarmos da posição do mouse para outro caso, teremos que criar um novo componente (ou seja, outro `<MouseWithCat>`) que renderiza algo especificamente para esse caso.
```

}
```

Agora, no lugar de clonar o componente `<Mouse>` e escrever o método `render` para resolver um caso específico, nós passamos uma _render prop_ que `<Mouse>` pode usar para determinar o que ele renderiza dinamicamente.

Portanto, **uma _render prop_ é uma função passada nas _props_ que um componente utiliza para determinar o que renderizar.**

> Essa técnica torna o comportamento que precisamos compartilhar extremamente protátil. Para acessar esse comportamento, basta renderizar um `<Mouse>` com uma `render` _prop_ que dirá o que renderizar com o (x, y) atual do cursor.

Reviewer: ```suggestion
Essa técnica torna o comportamento que precisamos compartilhar extremamente portátil. Para acessar esse comportamento, basta renderizar um `<Mouse>` com uma `render` _prop_ que dirá o que renderizar com o (x, y) atual do cursor.
```

</Mouse>
```

Você pode ver essa técnica sendo usada na API [react-motion](https://github.com/chenglou/react-motion). 

Dado que essa é uma técnica um pouco incomum, quando você estiver criando uma API como essa, você provavelmente vai querer definir nos seus `propTypes` que `children` deve ser uma função.

```js
Mouse.propTypes = {
  children: PropTypes.func.isRequired
};
```

> ## Avisos {#caveats}

Reviewer 2: Acho que poderiamos adicionar no glossario.

### Tenha cuidado ao utilizar _Render Props_ num React.PureComponent {#be-careful-when-using-render-props-with-reactpurecomponent}

> Usar uma _render prop_ pode anular a vantagem de utilizar [`React.PureComponent`](/docs/react-api.html#reactpurecomponent) se você criar uma função dentro de um método `render`. Isso se deve à comparação superficial de _prop_ sempre retornar `false` para novas _props_, e, nesse caso, cada `render` vai gerar um novo valor para a _render prop_.

Reviewer 2: Esse é um dos casos que eu me pergunto se não valeria a pena colocar o termo em ingles também, para facilitar a googlada marota.

Author: Qual seria o termo? `Shallow Prop Comparison`?

}
```

> Nos casos onde você não pode definir a prop estaticamente (e.g. porque você precisa esconder as props e o estado do componente), `<Mouse>` deveria estender `React.Component`.

Reviewer: ```suggestion
Nos casos onde você não pode definir a prop estaticamente (por exemplo, quando você precisa esconder as props e o estado do componente), `<Mouse>` deveria estender `React.Component`.
```
